### PR TITLE
[G2M] import styles component with a capital letter

### LINKED ISSUE #189: DOM error on imported packages
React emits the ` React does not recognize the prop on a DOM element` warning which is coming from the components imported with a camelcase name.

Screen Shot 2022-09-21 at 2 30 02 PM

### PR DESCRIPTION
**Pull request by parkdoeui**
Fixes #189

### CODE REVIEW

**Comment by geoerika** [comment]
@DoParkEQ Does this actually fix the warnings that you are looking to fix? I thought the fix was to change the names to lower case for the props that react complains about. I actually tested your branch in widget-studio and I still get the warnings:
Screen Shot 2022-09-22 at 11 00 46 AM

**Review by geoerika — CHANGES_REQUESTED** [review]
The fix is to rename those props as:
rightAlign -> rightalign
legendPosition -> legendposition
showLegend -> showlegend
in plotly files. No need to change anything else.

**Comment by DoParkEQ** [comment]
@geoerika It is still recommended to have react component's name with capital letter in order to distinguish them from DOM element. There is a possibility that it might cause unexpected quirks.

For further reference:
https://reactjs.org/docs/jsx-in-depth.html#user-defined-components-must-be-capitalized

**Comment by DoParkEQ** [comment]
@geoerika I encounter with `Field 'browser' doesn't contain a valid alias configuration` error when I link `chart-system` to `widget-studio`. Have you run into the same issue before?

**Comment by geoerika** [comment]
@DoParkEQ do you still want me to review this? I tested on my side and it's not solving the issue. I am confused of what you want me to review as I see the same solution as before.
I tested your solution creating locally a package from your local branch and installing it locally in widget-studio, and it's not working.

**Comment by DoParkEQ** [comment]
@geoerika of course I want you to review the change. For some reason the last change was not properly pushed to remote branch by the time I tagged you. Now you can check them out.

> The fix is to rename those props as:
> rightAlign -> rightalign
> legendPosition -> legendposition
> showLegend -> showlegend

This doesn't seem like a reliable approach. The real issue is coming from Reacts prop being applied to raw html element. Changing the prop names to lower case would be bypassing the actual problem. Initially, I thought it is due to the way it is imported, but the root cause is that `styled-component` directly passes the props to the curried element.

In terms of name format, it is recommended to have the react component's name with the uppercase letter. If you need further reference, feel free to check [this document](https://reactjs.org/docs/jsx-in-depth.html#user-defined-components-must-be-capitalized).

Hope this makes sense

File: src/components/plotly/shared/div-base.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import PropTypes from 'prop-types'
```
Comment (geoerika) [inline src/components/plotly/shared/div-base.js:1]: No need to create this component.

File: src/components/plotly/shared/styles/index.js [hunk 2]
```diff
@@ -4,14 +4,15 @@ import { setup, styled } from 'goober'
 import plotStyles from './plot'
 import legendStyles from './legend'
 import utilStyles from './util'
+import DivBase from '../div-base'
 
 setup(createElement)
```
Comment (geoerika) [inline src/components/plotly/shared/styles/index.js:9]: To be able to work with camel case props in goober, we can use one of the 2 mechanisms forwardProps, or shouldForwardProp:
https://goober.js.org/api/setup/#:~:text=With%20forwardProps-,%23,-The%20forwardProps%20function

**Review by geoerika — CHANGES_REQUESTED** [review]

**Comment by geoerika** [comment]
> @geoerika of course I want you to review the change. For some reason the last change was not properly pushed to remote branch by the time I tagged you. Now you can check them out.
> 
> > The fix is to rename those props as:
> > rightAlign -> rightalign
> > legendPosition -> legendposition
> > showLegend -> showlegend
> 
> I am worried that the suggested approach affects the consistency around naming convention as well as the component functionality. Technically those props are strictly used for styling therefore none of them should be passed to the actual DOM element as props. Changing the prop names to lower case would only mask the actual problem. Initially, I thought it is due to the way it is imported, but it seems that `styled-component` passes the props to the curried element.
> 
> In terms of formatting, it is recommended to have the react component's name with the uppercase letter. If you need further reference, feel free to check [this document](https://reactjs.org/docs/jsx-in-depth.html#user-defined-components-must-be-capitalized).
> 
> Let me know if you have any questions!

Fair enough. Although the problem here is not `styled-components` (we don't use it) as we use `goober` which is jsx syntax agnostic. I went bak to `goober` docs and found the way it handles these situations. Let's use that instead.

File: src/components/plotly/shared/styles/legend.js [hunk 3]
```diff
@@ -37,18 +36,22 @@ export default {
...
 
-  LegendColorBox: styled('div')(({ color, rightAligned }) => ({
-    order: + !rightAligned,
-    background: color,
-    borderRadius: '0.2rem',
-    width: '1rem',
-    height: '1rem',
-    margin: '0 0.5rem',
-  })),
+  LegendColorBox: styled('div')(({ rightAligned, color }) => {
```
Comment (geoerika) [inline src/components/plotly/shared/styles/legend.js:39]: @parkdoeui Not sure why we changed syntax here. The above should have worked perfectly and it is simplified. I prefer you not change it.

**Review by geoerika — APPROVED** [review]
@parkdoeui Reverse those 2 last changes please, otherwise it looks good.

File: src/components/plotly/shared/styles/legend.js (hunk 3 above)
Comment (parkdoeui) [reply src/components/plotly/shared/styles/legend.js:39]: I don't see any syntax change other than replacing `DivBase` to `div`. Can you please be more specific on what part of syntax are you referring to?

Comment (parkdoeui) [reply src/components/plotly/shared/styles/legend.js:39]: That was from the test whether the destructured objects cause the issue. Let me revert them

**Pull request merged**